syntax = "proto3";

message O1MeasurementReport {
  O1UEMeasurementReport ue_msg   = 1;
  O1GNBMeasurementReport gnb_msg  = 2;
  O1ULSCHFailure ulf  = 3;
  O1RRCFailure rrcf = 4;
  
}

message O1UEMeasurementReport {
  string imsi                   = 1;
  int32 rx_power_avg            = 2;
  int32 rx_power_tot            = 3;
  int32 n0_power_avg            = 4;
  int32 n0_power_tot            = 5;
  int32 rx_rssi_dBm             = 6;
  int32 ssb_rsrp_dBm            = 7;
  int32 mcs                     = 8;
}

message O1GNBMeasurementReport {
  int32 rnti                    = 1;
  int32 avg_rsrp                = 2;
  int32 srs_wide_band_snr       = 3;
  int32 dlsch_mcs               = 4;
  int32 ulsch_mcs               = 5;
  int32 cqi                     = 6;
  float dlsch_bler              = 7;
  float ulsch_bler              = 8;
}

message O1ULSCHFailure{
  int32 rnti                    = 1;
  int32 failure                 = 2;
}

message O1RRCFailure{
  int32 rnti                    = 1;
  int32 failure                 = 2;
}

